ocaml/xapi/xapi_db_upgrade.ml

```diff
@@ -506,6 +507,21 @@ let upgrade_vswitch_controller = {
         ignore (Xapi_sdn_controller.introduce ~__context ~protocol:`ssl ~address ~port:6632L)
 }
 
+let default_vm_platform_device_model = {
+  description = "Initialising unset VM.platform.device-model profiles";
+  version = (fun x -> x < jura);                             (* initialise only for upgrades from previous versions before jura *)
+  fn = fun ~__context ->
+    Db.VM.get_all ~__context
+    |> List.iter (fun vm ->
+      Db.VM.set_platform ~__context ~self:vm ~value:(
```

Comment:
I'd reorder this as:

```ocaml
Db.VM.get_platform ~__context ~self:vm
|> Xapi_vm_helpers.ensure_device_model_profile_present ~__context
   ~hVM_boot_policy:(Db.VM.get_HVM_boot_policy ~__context ~self:vm)
   ~default_value:Vm_platform.fallback_device_model_stage_1 (* fallback device model profile from previous version before jura *)
|> fun value ->
   Db.VM.set_platform ~__context ~self:vm ~value

Reply:
I like it. done in https://github.com/xapi-project/xen-api/pull/3275/commits/3d1bd6fb1adc0ad23097bedd9e23f027c80e2311